include/oneapi/tbb/flow_graph.h

```diff
@@ -1898,10 +1898,11 @@ class limiter_node : public graph_node, public receiver< T >, public sender< T >
             spin_mutex::scoped_lock lock(my_mutex);
             if( delta > 0 && size_t(delta) > my_count )
                 my_count = 0;
-            else if( delta < 0 && size_t(delta) > my_threshold - my_count )
+            else if( delta < 0 && delta > int(my_count - my_threshold) )
                 my_count = my_threshold;
             else
                 my_count -= size_t(delta); // absolute value of delta is sufficiently small
+            __TBB_ASSERT(my_count <= my_threshold, "counter values are truncated to be inside the [0, threshold] interval");
```

Comment:
Make the assert be read as not related to the last `else` branch?
```suggestion
            else {
                my_count -= size_t(delta); // absolute value of delta is sufficiently small
            }
            __TBB_ASSERT(my_count <= my_threshold, "counter values are truncated to be inside the [0, threshold] interval");
```